src/utils/flagging.R

```diff
@@ -173,13 +208,32 @@ calculate_flags <- function(
   # check when we had a day with value above x
   # and 0 in the previous # number of dates
   if (!null_fs) {
-    first_since <- sort(first_since, decreasing = FALSE)
-    for (fs_period in first_since) {
-      .data <- dplyr$mutate(
-        .data,
-        "flag_first_{{ fs_period }}" := .data[[x_col]] > 0 & dplyr$lag(.data[[paste0("rs_", fs_period)]]) == 0
+    # create the forward looking 60 day period for checking the minimums
+    .data <- dplyr$mutate(
+      .data,
+      rs_next_60 = zoo$rollsum(
+        x = .data[[x_col]],
+        k = 60,
+        fill = "extend",
+        align = "left"
       )
-    }
+    )
+
+    df_fs <- purrr$map2(
+      .x = first_since,
+      .y = first_since_minimums,
+      .f = \(fs_period, fs_minimum) {
+        dplyr$mutate(
+          .data,
+          "flag_first_{{ fs_period }}" := .data[[x_col]] > 0 & dplyr$lag(.data[[paste0("rs_", fs_period)]]) == 0 & rs_next_60 >= fs_minimum
+        ) |>
+          dplyr$ungroup() |>
+          dplyr$select(paste0("flag_first_", fs_period))
+      }
+    ) |>
+      dplyr$bind_cols()
+
+    .data <- dplyr$bind_cols(.data, df_fs)
   }
```

Comment:
what do you think about using  `purrr$map2_dfc()` & leaving `iso3` and `date` so that we can the `left_join(.data,df_fs, by =c("iso3","date"))` -- this might provide some guard rails against anything strange happening?

Reply:
Thanks, good point. I have added a comment to clarify why I did it this way, which is essentially, we ensure that you don't have to have `iso3` or `date` columns or ANY specific columns, and if you have additional columns they won't be lost. We simply add only the new columns.

Reply:
Which I think makes it robust to strange behavior in case you passed something in with extra columns that might get lost for instance.